### PR TITLE
refactor: make dynamic wave effects can be disabled by token.motion

### PR DESCRIPTION
**Pull request by PhosphorusP**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/44084

### 💡 Background and solution

The transition duration attributes in components/wave/style.ts is hard-coded and cannot respond to different token.motion values. To solve this problem, modified style in components/wave/style.ts.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Make dynamic wave effects can be disabled by token.motion      |
| 🇨🇳 Chinese |     使波纹效果可被token.motion关闭      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 67fb716</samp>

Reduce the motion of the wave effect for accessibility. Use `token.motion` to control the transition duration of `box-shadow` and `opacity` in `components/_util/wave/style.ts`.

### <samp>🤖 Generated by Copilot at 67fb716</samp>

*  Modify the transition duration of the wave effect to depend on the `token.motion` value, which allows users to disable the motion effect for accessibility reasons ([link](https://github.com/ant-design/ant-design/pull/44087/files?diff=unified&w=0#diff-c9ba011ec30b589636721e4ffe4cb109abc1341f6f32f7d16881f45ecd9b2caeL24-R25), [link](https://github.com/ant-design/ant-design/pull/44087/files?diff=unified&w=0#diff-c9ba011ec30b589636721e4ffe4cb109abc1341f6f32f7d16881f45ecd9b2caeL35-R36))

### CODE REVIEW

File: components/_util/wave/style.ts [hunk 1]
```diff
@@ -32,8 +32,8 @@ const genWaveStyle: GenerateStyle<WaveToken> = (token) => {
 
         '&.wave-quick': {
           transition: [
-            `box-shadow 0.3s ${token.motionEaseInOut}`,
-            `opacity 0.35s ${token.motionEaseInOut}`,
+            `box-shadow ${token.motion ? 0.3 : 0}s ${token.motionEaseInOut}`,
+            `opacity ${token.motion ? 0.35 : 0}s ${token.motionEaseInOut}`,
```
Comment (afc163) [inline components/_util/wave/style.ts:36]: It is better to disable motion in https://github.com/ant-design/ant-design/blob/master/components/_util/wave/index.ts

File: components/_util/wave/style.ts [hunk 2]
```diff
@@ -21,8 +21,8 @@ const genWaveStyle: GenerateStyle<WaveToken> = (token) => {
       // =================== Motion ===================
```
Comment (captain-hoity-toity) [inline components/_util/wave/style.ts:21]: Could we extract this hardcoded string into a configuration or constant?

File: components/_util/wave/style.ts (hunk 1 above)
Comment (afc163) [reply components/_util/wave/style.ts:36]: ping

Comment (afc163) [reply components/_util/wave/style.ts:36]: ping

**Pull request closed without merging**